docs/synapse/devguides/power-ups.rst

```diff
@@ -0,0 +1,426 @@
+        Use !quit to exit.
+        Use !help to see local interpreter commands.
+
+    storm> acme.hello.sayhi
+    hello storm!
+    complete. 0 nodes in 1 ms (0/sec).
+    storm>
+
+Directory Layout
+----------------
+
+Storm Modules
+=============
+
+Deploying **Storm** modules allows you to author powerful library functions that you can use in
+automation or **Storm** commands to facilitate code re-use and enforce privilidge separation boundaries.
+
+A **Storm** module is specified within the ``modules:`` section of the **Storm Package** YAML file.
+
+::
+    modules:
+
+      - name: acme.hello.example00
+        modconf:
+            varname: varvalu
+            othervar: [1, 2, 3]
+
+The ``modconf:`` key can be used to specify variables which will be mapped into the module's **Storm**
+runtime and accessible using the implicit variable ``$modconf``::
+
+    function foo() {
+        $lib.print($modconf.varname)
+        return((10))
+    }
+
+    function bar() {
+        for $i in $modconf.othervar {
+            // do something using $i...
+        }
+    }
+
+Priviledged Modules
+-------------------
+
+In order to facilitate delegating permission for priviledged operations, **Storm** modules may specify
+permissions which allow the module to be imported with admin priviledges. It is a best-practice to declare
+these permissions within the **Storm** package using the ``perms:`` key before using them::
+
+    perms:
+      - perm: [ acme, hello, user ]
+        gate: cortex
+        desc: Allows a user to call priviliedged APIs from Acme-Hello.
+
+    modules:
+
+      - name: acme.hello.privsep
+        asroot:perms:
+            - [ acme, hello, user ]
+
+To minimize risk, you must very carefully consider what functions to implement within a priviledged **Storm**
+module! Priviledged modules should contain the absolute minimum required functionality.
+
+An excellent example use case for a priviledged **Storm** module exists when you have an API key or password
+which you would like to use on a user's behalf without disclosing the actual API key. The **Storm** library
+``$lib.globals.set(<name>, <valu>)`` and ``$lib.globals.get(<name>)`` can be used to access protected global
+variables which regular users may not access without special permissions.  By implementing a priviledged
+**Storm** module which retrieves the API key and uses on the user's behalf without disclosing it, you may
+protect the API key from disclosure while also allowing users to use it. For example,
+``acme.hello.privsep.storm``::
+
+    function getFooByBar(bar) {
+
+        $apikey = $lib.globals.get(acme:hello:apikey)
+        $headers = ({"apikey": $apikey})
+
+        $url = $lib.str.format("https://acme.newp/api/v1/foo/{bar}", bar=$bar)
+
+        $resp = $lib.inet.http.get($url, headers=$headers)
+        if ($resp.code != 200) {
+            $lib.warn("/api/v1/foo returned HTTP code: {code}", code=$resp.code)
```

Comment:
Minor nit: probably want to return null here or show an example of try/catch to attempt to get a json error message.